react/OrderItems.tsx

```diff
@@ -0,0 +1,112 @@
+import { adjust } from 'ramda'
+import React, {
+  createContext,
+  FunctionComponent,
+  ReactNode,
+  useContext,
+  useState,
+} from 'react'
+import { branch, renderComponent } from 'recompose'
+import { compose, graphql } from 'react-apollo'
+import { useOrderManager } from 'vtex.order-manager/OrderManager'
+
+import ItemList from './graphql/itemList.graphql'
+import UpdateItem from './graphql/updateItem.graphql'
+
+interface Context {
+  itemList: Item[]
+  updateItem: (index: number, quantity: number) => void
+  loading: boolean
+}
+
+interface OrderItemsProviderProps {
+  children: ReactNode
+  ItemListQuery: any
+  UpdateItem: any
+}
+
+const OrderItemsContext = createContext<Context | undefined>(undefined)
+
+const EmptyState: FunctionComponent = ({ children }: any) => {
+  const updateItem = async (_: number, __: number) => {}
+  return (
+    <OrderItemsContext.Provider
+      value={{ itemList: [], updateItem, loading: true }}
```

Comment:
When filling context Provider value and its value it's an object, use useMemo to memoize the object.

```tsx
const value = useMemo(() => ({ itemList: [], updateItem, loading: true }), [])

return (
  <OrderItemsContext.Provider value={value}>
```